### DISCUSSION TITLE
home assistant posting data to emoncms through MQTT

### DISCUSSION THREAD

**Discussion by alexjunk** [Open-Building-Management/emoncms#16] [MQTT]
actually testing the ingress version, with emoncms running inside home assistant

i am not an advanced user of home assistant

I've noticed that some meteo datas were delivered from met.no, by the Norwegian Meteorological Institute

![image](https://github.com/Open-Building-Management/emoncms/assets/43913055/728fa6e3-a162-4dee-8fd9-92a2e58b9ece)

using the MQTT integration

![image](https://github.com/Open-Building-Management/emoncms/assets/43913055/684d3a30-ae62-4f3a-b9de-b34f01fc7c3e)

and an automation to publish some meteo data on emoncms

![image](https://github.com/Open-Building-Management/emoncms/assets/43913055/add8d2f1-8f60-4459-ab8f-996683c5793c)

it seems to work :-)

![](https://github.com/Open-Building-Management/emoncms/assets/43913055/260809c8-e525-4543-a3da-dbb5d8f49226)

**Comment by ashipaek0** [comment]
Hello.

I noticed you were able to send HA MQTT to emoncms.


I have a Raspberry Pi publishing    prefix solar_assistant/#   on my local network at 192.168.1.101

i am not sure what to put in the host box and port  in emoncms

**Comment by alexandrecuer** [comment]
@ashipaek0 : the addon is listening on topic emon, so you should publish on that topic. If you are running out of homeassistant with docker compose for example, you can change the topic with an env var named MQTT_BASETOPIC  but you cant do that in the home assistant addon

For the ip/port : 
1) if you use the internal broker of the addon, it should be the ip of the machine running the addon and port 9883 if you did not customise the settings...
2) if you use an external broker (it should be your case) you must use port 1883, so try 192.168.1.101 and port 1883, and also tune your broker as said above to publish on topic `emon`

**Reply by ashipaek0** [reply]
Thank you for your response.

I have tried 

192.168.1.101:9883      in the host box, but the MQTT does not appear in emoncms

the HomeAssistant box is at 192.168.1.180

and i have also tried 

192.168.1.180:9883      in the host box, without success

For clarity the MQTT data is in Home Assistant right now.  i have dashboards with the data.

**Reply by ashipaek0** [reply]
I should also mention that my install does not actually have a box for the port.
![Screenshot from 2024-04-27 14-31-45](https://github.com/Open-Building-Management/emoncms/assets/25457085/f268f81b-d101-4e3f-a25a-17e38fa7d0cd)



i notice that difference from the screenshots aboove

**Reply by alexandrecuer** [reply]
Just try 192.168.1.101 in the host box. But your broker should publish on topic emon not solar_assistant

And check the emoncms logs in the admin module of emoncms

When the broker is connected, you should see something like that :
```
2024-04-17 13:36:05.-350|WARN|emoncms_mqtt.php|Not connected, retrying connection
2024-04-17 13:36:05.-339|WARN|emoncms_mqtt.php|Connecting to MQTT server: Connection Accepted.: code: 0
```

**Reply by ashipaek0** [reply]
yeah i think i understand it now.

unfortunately changing the broker is going to break a lot of other stuff so that's not an option.
the other approach would be to change the emoncms default MQTT parameters.

i see that in the dockerfile the MQTT parameters for emoncms are set, so my attempts at getting into the container itself and editing settings.ini does not work.

anyway, thank you for your time and assistance.

**Reply by alexandrecuer** [reply]
If you have the data (as states) already in homeassistant you can route them back to emoncms using the technic above : https://github.com/Open-Building-Management/emoncms/discussions/16#discussion-6520812

First activate the MQTT integration and configure it with the ip of the addon, 192.168.1.180 in your case. Restore the MQTT conf of the addon with the standard parameters, localhost for the host, and username / password as you like
The port exposed is in the network section of the conf, you can keep 9883

Then test if you can post to emoncms with the MQTT integration : 
![](https://github.com/Open-Building-Management/emoncms/assets/24553739/418e0516-dc3c-409b-bf92-57f11dc43d9c)

After you should be able to reproduce what i did for the meteo forecast with your solar datas.....

I can adapt the addon with a basetopic settings if this is not working, but i think it should. At the beginning, it was quite tedious to work with the MQTT integration and to find the state names 
but finally it was ok so i think it is also possible in your case

**Reply by ashipaek0** [reply]
Thanks.

I did find that one could place mosquitto config files in /share/mosquitto     in Home Assistant

so i put in a bridge.conf

```
connection bridge01
address 192.168.1.101:1883
remote_username emonpi
remote_password emonpimqtt2016
topic # both 0  emon/ solar_assistant/
```

and after a restart, the data showed up in emoncms inputs.

so i guess thats a solution


but i do believe that it would be neater to have the basetopic settings in the addon.


Thanks

**Reply by alexandrecuer** [reply]
@ashipaek0 : new version `alpine3.19_emoncms11.5.0_1` available with the ability to tune the mqtt topic in the addon config. It should solve your problem without the need for a bridge. Let me know if it works for you.

Anyway the bridge solution is nice :-) I'll keep it in mind and thanks for sharing. I assume you're referring to this  : https://solar-assistant.io/help/home-assistant/broker-setup ?

**Reply by ashipaek0** [reply]
Yes indeed. That's the writeup that gave me the clue.

Thanks for the new build 

Yes the base topic does solve the issue 

Great
